### ISSUE TITLE
UI doesn't handle Akismet keys set via constant properly

### ISSUE THREAD

**Issue by beaulebens** [Automattic/jetpack#8226]
Sites using Akismet may optionally configure their API Key via a constant (`WPCOM_API_KEY`, for historical reasons). This is common on hosts who have an Enterprise license for Akismet usage, and provide it for all of their users. Pressable is an example of this. It might also been seen for example on a multi-site, where the site admin sets it up for all sub-sites. In cases where Akismet is configured in this way, our UIs in both the Jetpack plugin, and Calypso/WP.com handle it poorly.

#### Steps to reproduce the issue
- create a WordPress site, install, activate, connect Jetpack
- get an Akismet API key
- if your site has it, delete any value stored in `wp_options` under the name `wordpress_api_key`
- add `define('WPCOM_API_KEY', 'key-here');` to your `wp-config.php`, where 'key-here' is your Akismet API key
- go to _Jetpack > Settings >Security_ and see that the "Spam filtering" section thinks you need an API key, even though you technically have one.
- now go to wordpress.com and for that site, look under Settings > Security and see that it's pretty broken there as well.

#### What I expected
We should probably detect (will require a flag in the API) that the key is coming from a constant, is thus not editable, and just display a read-only message that things are configured, but they can't change it from here.

#### What happened instead
In Jetpack/wp-admin:
![screen shot 2017-11-21 at 3 18 43 pm](https://user-images.githubusercontent.com/108942/33101641-47aa2186-ced6-11e7-9c16-952925e20a02.png)

In WP.com:
![screen shot 2017-11-21 at 3 19 58 pm](https://user-images.githubusercontent.com/108942/33101654-5293be4a-ced6-11e7-8c24-0be0a1618f2c.png)

**Comment by BandonRandon** [comment]
The UI still shows a key is required, except now the settings show up in red that a key is needed:
![screenshot_10-07-2018-13 38 39](https://user-images.githubusercontent.com/150348/42536386-c634e678-8446-11e8-9ddb-3355cc12a014.png)

**Comment by mdmuzzie** [comment]
We encounter this issue as well, and we would love a fix. Jetpack does not recognize our Akismet Enterprise API key that is configured in our wp-config.php file.

**Comment by desnum** [comment]
A user hopped into chat to ask about this: Akismet is already activated but the text "Your site needs an Antispam key." is confusing. 

hc-12641715

**Comment by JessBoctor** [comment]
I had another user confused by this in chat. This was especially confusing for them because they have multiple sites and wasn't sure if they needed an API key for each of their sites, or only one, and if they needed to create more accounts for their API keys.

18906850-HC

**Comment by eeeeevon13** [comment]
Let me know if I'm understanding the problem correctly here. It seems as though there are a few parts to this issue:

1) Users confused if they need multiple API keys for multiple sites
2) Jetpack is not recognizing the enterprise API key set if set in wp-config file (or other sources?)

@beaulebens 
```
if your site has it, delete any value stored in wp_options under the name wordpress_api_key
add define('WPCOM_API_KEY', 'key-here'); to your wp-config.php, where 'key-here' is your Akismet API key
```

**Couple of questions:**

1. @beaulebens I see you've given directions on how to recreate from a technical perspective, can you explain how a user might actually get into this state? When/how might they recreate this condition via the UI?
2. I've added a mock that is geared toward this idea that the API key was set directly in wp-config, are there other places a user can set the API key that might also create this state?
3. Does a user need multiple API keys? @JessBoctor 

**Suggested action items:**

1. Add a flag in order to understand that API is valid and coming from a different source
2. Add a user state to explain this to me including "locked field" state and messaging
![lockedfield](https://user-images.githubusercontent.com/49159284/84074918-f9d79600-a998-11ea-831b-00b8cb1639e5.png)

**Comment by BrookeDot** [comment]
Often when a key is set in `wp-config.php` it is done so as the provider, often the host, has set an Enterprise key.

 I like your mockup, but I think it may be wise to hide the actual key. "A valid key has been set in wp-config.php" and that message can then be hookable in Jetpack allowing hosts to change that as they see fit.

**Comment by beaulebens** [comment]
Yep I think I agree with @BrookeDot's suggestion about hiding the key -- maybe just a big checkmark/"Your antispam protection is enabled" type message, rather than showing them the key as well.

The specific problem here is when a key is already set in a config file, or via some other code-based approach, so it's not really something that a normal user can do to themselves _purely via the UI_.

It sounds like there's maybe some (understandable) confusion around multiple keys -- you can only easily use one Akismet API key at a time (all Akismet API requests require a key). For strange legacy reasons, that key is named `WPCOM_API_KEY` when configured via code (but it's actually an Akismet key).

So -- if a key is configured via code, you don't need/shouldn't be able to set one via the UI, and the key being used should probably be hidden from view.

**Comment by BrookeDot** [comment]
> if a key is configured via code, you don't need/shouldn't be able to set one via the UI, and the key being used should probably be hidden from view.

I don't know what it would look like, but perhaps changing the field type to `password` might also work to keep with the original mockup.

**Comment by BrookeDot** [comment]
Looks good! my one bit of minor feedback is since `WPCOM_API_KEY` could be set outside of `wp-config.php` (although not common) I'd suggest we don't put too much emphasis on the location itself. 

What about:
> It looks like your API key has been set globally

> A key has been set in your site's configuration.

**Comment by mattgawarecki** [comment]
I'm working on a fix for this. Slow going so far, but going nonetheless 😉

**Comment by mattgawarecki** [comment]
I've got a potential fix in #16542, as GitHub indicates above. All reviews welcome, but I definitely want to make sure we're happy with the design since there was a little iteration going on back in June. 🙂 

cc @BrookeDot @eeeeevon13
